### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by julia-podrazka**

### CODE REVIEW

File: crates/cargo-bsp/src/project_model/cargo_package.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//! Stores necessary information that can be obtained about Cargo package with
+//! available and enabled features (relevant for the feature requests
+//! that are not yet added to the BSP).
```
Comment (Toomimi) [inline crates/cargo-bsp/src/project_model/cargo_package.rs:3]: Może zmieniłbym na (relevant for the Cargo extension for BSP, which allows toggling the features, not yet added to the BSP documentation)

File: crates/cargo-bsp/src/project_model/workspace.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+//! Discovers project's workspace based on `cargo metadata` command.
```
Comment (Toomimi) [inline crates/cargo-bsp/src/project_model/workspace.rs:1]: Tutaj dodałbym, że to jest właśnie to co przechowujemy w stanie naszego serwera

File: crates/cargo-bsp/src/project_model/project_manifest.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//! Discovers Cargo project manifest.
```
Comment (Toomimi) [inline crates/cargo-bsp/src/project_model/project_manifest.rs:1]: Cargo project's manifest discovery, in order to call Cargo metadata properly, up2u

File: crates/cargo-bsp/src/project_model/package_dependency.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+//! Stores package dependencies.
```
Comment (Toomimi) [inline crates/cargo-bsp/src/project_model/package_dependency.rs:1]: Structure representing a dependency of a package

Comment (Toomimi) [reply crates/cargo-bsp/src/project_model/package_dependency.rs:1]: Fields _features and _uses_default_features may become handy when implementing BuildTargetDependencyModules request. Optional dependencies of the PackageDependency are included only if specific features for PackageDepenency are set. Similarly, disabling default features impact the set of dependencies of the PackageDependency.

File: crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs [hunk 5]
```diff
@@ -1,3 +1,24 @@
+//! CreateCommand trait implementation for the Compile/Run/TestParams.
+//! The trait allows getting origin id and creating commands regardless if it is the compile,
+//! run or test request.
+//!
+//! There are two types of commands:
+//! - requested: standard `cargo build`, `cargo run` and `cargo test` to compile,
+//! run and test the project,
+//! - unit graph: the same as before but with `--unit-graph -Z unstable-options` flags
+//! (only available with `+nightly`). These commands are used to get the amount of steps
+//! during the compilation for the command executed without this flag.
```
Comment (agluszak) [inline crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs:10]:
```suggestion
//! (only available with `+nightly`). These commands are used to get the number of compilation steps.
```

File: crates/cargo-bsp/src/cargo_communication/request_actor.rs [hunk 6]
```diff
@@ -1,3 +1,25 @@
+//! Handles messages from Cargo command, parsing them and preparing appropriate
+//! notifications and responses. Also handles information about the finish of
+//! Cargo command and the cancel request from the client.
+//!
+//! The hierarchy of tasks:
```
Comment (agluszak) [inline crates/cargo-bsp/src/cargo_communication/request_actor.rs:5]: super że to jest opisane <3

File: crates/cargo-bsp/src/cargo_communication/request_actor.rs [hunk 7]
```diff
@@ -170,6 +189,7 @@ where
         }
     }
 
+    // Cancel should cancel all started tasks, not only the root task.
```
Comment (agluszak) [inline crates/cargo-bsp/src/cargo_communication/request_actor.rs:192]:
```suggestion
    // TODO: Cancel should cancel all started tasks, not only the root task.
```

File: crates/cargo-bsp/src/cargo_communication/request_actor_state.rs [hunk 8]
```diff
@@ -48,7 +53,10 @@ pub struct TestState {
     pub(super) task_id: TaskId,
     pub(super) suite_task_id: TaskId,
     pub(super) suite_task_progress: SuiteTaskProgress,
+    // Currently tested build target.
     pub(super) current_build_target: Option<BuildTargetIdentifier>,
+    // Maps single tests name (by which they are recognized by Cargo) to the TaskId
+    // of the task that they started.
```
Comment (agluszak) [inline crates/cargo-bsp/src/cargo_communication/request_actor_state.rs:59]:
```suggestion
    /// Currently tested build target.
    pub(super) current_build_target: Option<BuildTargetIdentifier>,
    /// Maps single tests name (by which they are recognized by Cargo) to the TaskId
    /// of the task that they started.
```

File: crates/cargo-bsp/src/project_model/workspace.rs [hunk 9]
```diff
@@ -1,3 +1,7 @@
...
+//! `ProjectWorkspace` is stored in the state of the server.
+//! Creates hashmaps for easy access to the project's information.
```
Comment (agluszak) [inline crates/cargo-bsp/src/project_model/workspace.rs:3]: nie rozumiem trochę co to znaczy

File: crates/cargo-bsp/src/server/handlers.rs [hunk 10]
```diff
@@ -40,50 +43,47 @@ pub(crate) fn handle_sources(
     })
 }
 
+// Not properly handled yet
```
Comment (agluszak) [inline crates/cargo-bsp/src/server/handlers.rs:46]:
```suggestion
// TODO: Not properly handled yet
```

File: crates/cargo-bsp/src/server/handlers.rs [hunk 11]
```diff
@@ -40,50 +43,47 @@ pub(crate) fn handle_sources(
...
-// the current version of the client sends a java extension request even though we are not working with java.
-// to be removed once it is fixed in the client
-pub(crate) fn handle_java_extensions(
-    _: GlobalStateSnapshot,
-    _: bsp_types::requests::JavacOptionsParams,
-) -> Result<bsp_types::requests::JavacOptionsResult> {
-    Ok(bsp_types::requests::JavacOptionsResult::default())
-}
-
+// Not properly handled yet
```
Comment (agluszak) [inline crates/cargo-bsp/src/server/handlers.rs:54]:
```suggestion
// TODO: Not properly handled yet
```

File: crates/cargo-bsp/src/server/handlers.rs [hunk 12]
```diff
@@ -40,50 +43,47 @@ pub(crate) fn handle_sources(
...
 pub(crate) fn handle_clean_cache(
     _: GlobalStateSnapshot,
     _: bsp_types::requests::CleanCacheParams,
 ) -> Result<bsp_types::requests::CleanCacheResult> {
     Ok(bsp_types::requests::CleanCacheResult::default())
 }
 
+// Not properly handled yet
```
Comment (agluszak) [inline crates/cargo-bsp/src/server/handlers.rs:62]: itd.

File: crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs (hunk 5 above)
Comment (julia-podrazka) [reply crates/cargo-bsp/src/cargo_communication/cargo_types/cargo_command.rs:10]: Done

File: crates/cargo-bsp/src/project_model/workspace.rs (hunk 9 above)
Comment (julia-podrazka) [reply crates/cargo-bsp/src/project_model/workspace.rs:3]: Przeformułowałam

**Review by agluszak — APPROVED** [review]

**Pull request merged**